x-pack/elastic-agent/pkg/agent/program/testdata/single_config.yml

```diff
@@ -19,37 +19,49 @@ outputs:
     hosts: ["monitoring:9200"]
     ca_sha256: "7lHLiyp4J8m9kw38SJ7SURJP4bXRZv/BNxyyXkCcE/M="
 
-datasources:
-  - use_output: default
-    inputs:
-    - type: docker/metrics
-      streams:
-        - metricset: status
-          dataset: docker.status
-        - metricset: info
-          dataset: ""
-      hosts: ["http://127.0.0.1:8080"]
-    - type: logs
-      streams:
-        - paths:
-          - /var/log/hello1.log
-          - /var/log/hello2.log
-      vars:
-        var: value
-  - namespace: testing
+inputs:
+  - type: docker/metrics
     use_output: default
-    inputs:
-      - type: apache/metrics
-        processors:
-          - add_fields:
-              fields:
-                should_be: first
-        streams:
-          - enabled: true
-            metricset: info
-            hosts: ["http://apache.remote"]
-        hosts: ["http://apache.local"]
-        id: apache-metrics-id
+    streams:
+      - metricset: status
+        dataset:
```

Comment:
I assume `dataset.name` and 

```
dataset:
  name: foo
```

work?

Reply:
yeah it was a bit tricky as transpiler understands `dataset.name` as a single key but i made both versions work